### PR TITLE
NATIVE-170: Mapping city model

### PR DESCRIPTION
**Pull request by Simple-codinger**

### CODE REVIEW

File: src/modules/endpoint/DatabaseConnector.js [hunk 1]
```diff
@@ -52,6 +53,15 @@ type ContentEventJsonType = {|
   |}
 |}
 
+type ContentCityJsonType = {|
+    name: string,
+    live: boolean,
+    code: string,
+    extrasEnabled: boolean,
+    eventsEnabled: boolean,
+    sortingName: string
```
Comment (klinzo) [inline src/modules/endpoint/DatabaseConnector.js]: `prefix` is missing here, was added some time ago

File: src/modules/endpoint/DatabaseConnector.js [hunk 2]
```diff
@@ -87,6 +98,10 @@ class DatabaseConnector {
   }
 
   getMetaCitiesPath (): string {
+    return `${CACHE_DIR_PATH}/metaCities.json`
```
Comment (maxammann) [inline src/modules/endpoint/DatabaseConnector.js]: By following the current naming schema this should be:
```suggestion
    return `${CACHE_DIR_PATH}/cities-meta.json`
```

File: src/modules/endpoint/DefaultDataContainer.js [hunk 3]
```diff
@@ -100,8 +100,7 @@ class DefaultDataContainer implements DataContainer {
     await cache.cache(categories, context)
   }
 
-  setCities = (cities: Array<CityModel>) => {
-    // TODO: Offline available cities will be persisted in NATIVE-175. For now switching cities when offline is not possible.
+  setCities = async (cities: Array<CityModel>) => {
     this._cities = cities
     return Promise.resolve()
```
Comment (maxammann) [inline src/modules/endpoint/DefaultDataContainer.js]: I think this should write the cities.json.

File: src/modules/endpoint/DefaultDataContainer.js [hunk 4]
```diff
@@ -100,10 +100,9 @@ class DefaultDataContainer implements DataContainer {
     await cache.cache(categories, context)
   }
 
-  setCities = (cities: Array<CityModel>) => {
-    // TODO: Offline available cities will be persisted in NATIVE-175. For now switching cities when offline is not possible.
+  setCities = async (cities: Array<CityModel>) => {
     this._cities = cities
-    return Promise.resolve()
+    await this._databaseConnector.storeCities(cities)
```
Comment (maxammann) [inline src/modules/endpoint/DefaultDataContainer.js]: I'm not 100% sure I think you have to return the promise. Else it is just resolved as soon as the function returns.
```suggestion
    return this._databaseConnector.storeCities(cities)
```

File: src/modules/endpoint/sagas/watchFetchCities.js [hunk 5]
```diff
@@ -14,11 +14,14 @@ import { baseUrl } from '../constants'
...
 function * fetchCities (dataContainer: DataContainer): Saga<void> {
   try {
-    const payload: Payload<Array<CityModel>> = yield call(() => request(createCitiesEndpoint(baseUrl)))
-
-    const cities: Array<CityModel> = payload.data
-
-    yield call(dataContainer.setCities, cities)
+    let cities: Array<CityModel>
+    if (yield call(() => dataContainer.citiesAvailable())) {
+      cities = yield call(() => dataContainer.getCities())
```
Comment (Schedulaar) [inline src/modules/endpoint/sagas/watchFetchCities.js]: The user should be able to force a refresh (using pull-to-refresh) of the cities.
If you want, you can implement that in this task or in another task.
If you choose the latter please create one for that :)

**Pull request merged**